packages/react/src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,276 @@
+import * as React from 'react'
+import * as _ from 'lodash'
+import * as PropTypes from 'prop-types'
+import * as customPropTypes from '@stardust-ui/react-proptypes'
+import { handleRef, Ref } from '@stardust-ui/react-component-ref'
+import cx from 'classnames'
+
+import {
+  applyAccessibilityKeyHandlers,
+  AutoControlledComponent,
+  ChildrenComponentProps,
+  commonPropTypes,
+  isFromKeyboard,
+  partitionHTMLProps,
+  UIComponentProps,
+  RenderResultConfig,
+  setWhatInputSource,
+} from '../../lib'
+import {
+  ComponentEventHandler,
+  ShorthandValue,
+  WithAsProp,
+  withSafeTypeForAs,
+  Omit,
+} from '../../types'
+import { Accessibility } from '../../lib/accessibility/types'
+import { sliderBehavior } from '../../lib/accessibility'
+import { SupportedIntrinsicInputProps } from '../../lib/htmlPropsUtils'
+import Box from '../Box/Box'
+
+const defaultPropValues = {
+  max: 100,
+  min: 0,
+  step: 1,
+  value: 50,
+}
+
+const processInputValues = (
```

Comment:
Can we do this inside `getDerivedStateFromProps()` (`getAutoControlledStateFromProps()`). In that case we will have more accurate values for accessibility and styles...

Reply:
does that mean `min`, `max` and `valueAsPercentage` will become part of state? why do we need that?
From accessibility POV it works anyway with both strings and numbers.

Are there any other concerns? It seems strange to add these to the state. @layershifter

Reply:
> does that mean `min`, `max` and `valueAsPercentage` will become part of state?

Yep

> From accessibility POV it works anyway with both strings and numbers.
> Are there any other concerns? It seems strange to add these to the state.

In this case: no.